server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailetContext.java

```diff
@@ -248,14 +257,27 @@ public boolean isLocalEmail(MailAddress mailAddress) {
                 return false;
             }
             try {
-                return localusers.contains(localusers.getUser(mailAddress));
-            } catch (UsersRepositoryException e) {
+                return isLocaluser(mailAddress)
+                    || isLocalAlias(mailAddress);
+            } catch (UsersRepositoryException | ErrorMappingException | RecipientRewriteTableException e) {
                 LOGGER.error("Unable to access UsersRepository", e);
             }
         }
         return false;
     }
 
+    private boolean isLocaluser(MailAddress mailAddress) throws UsersRepositoryException {
+        return localusers.contains(localusers.getUser(mailAddress));
+    }
+
+    private boolean isLocalAlias(MailAddress mailAddress) throws UsersRepositoryException, ErrorMappingException, RecipientRewriteTableException {
```

Comment:
This method would probably deserve its own class, JamesMailetContext is already bloated and does too many things, let's not make it bigger.